test/buffers_utest.cpp

```diff
@@ -63,222 +55,156 @@ class BuffersUtest {
-        std::vector<T*>* odd_mem = new std::vector<T*>(half, nullptr);
-        std::unique_ptr<quadiron::vec::Buffers<T>> i_even =
-            std::unique_ptr<quadiron::vec::Buffers<T>>(
-                new quadiron::vec::Buffers<T>(half, size, even_mem));
-        std::unique_ptr<quadiron::vec::Buffers<T>> i_odd =
-            std::unique_ptr<quadiron::vec::Buffers<T>>(
-                new quadiron::vec::Buffers<T>(half, size, odd_mem));
-        vec1->separate_even_odd(i_even.get(), i_odd.get());
-
-        // vec1->dump();
-        vec1->separate_even_odd();
-        // vec1->dump();
-
-        quadiron::vec::Buffers<T> _i_even(vec1.get(), 0, half);
-        quadiron::vec::Buffers<T> _i_odd(vec1.get(), half, n);
-        assert(i_even->eq(&_i_even));
-        assert(i_odd->eq(&_i_odd));
-
-        // vec2.dump();
-
-        std::vector<T*>* mem1 = vec1->get_mem();
-        std::vector<T*>* mem2 = vec2.get_mem();
-
-        bool ok = true;
-        for (i = 0; i < n / 2; i += 2) {
-            T* even1 = mem1->at(i);
-            T* even2 = mem2->at(i * 2);
-            T* odd1 = mem1->at(i + n / 2);
-            T* odd2 = mem2->at(i * 2 + 1);
-            for (j = 0; j < size; j++) {
-                if (even1[j] != even2[j] || odd1[j] != odd2[j]) {
-                    ok = false;
-                    i = n;
-                    break;
-                }
-            }
+    ASSERT_EQ(vec2.get_n(), end - begin);
+    ASSERT_EQ(vec2.get_size(), vec1->get_size());
+
+    for (int i = 0; i < end - begin; i++) {
+        for (int j = 0; j < size; j++) {
+            mem2->at(i)[j] = mem1->at(i + begin)[j];
         }
-        assert(ok);
     }
 
-    void buffers_utest3()
-    {
-        std::cout << "buffers_utest3\n";
-
-        int n = 8;
-        int size = 32;
-        int n1 = 4;
-        int n2 = 10;
-
-        quadiron::vec::Buffers<T>* vec = gen_buffers_rand_data(n, size);
-        quadiron::vec::Buffers<T> vec1(vec, n1);
-        quadiron::vec::Buffers<T> vec2(vec, n2);
-
-        quadiron::vec::Buffers<T> _vec1(vec, 0, n1);
-        quadiron::vec::Buffers<T>* _vec2 =
-            new quadiron::vec::BuffersZeroExtended<T>(vec, n2);
-
-        assert(vec1.eq(&_vec1));
-        assert(vec2.eq(_vec2));
-
-        delete vec;
-
-        quadiron::vec::Buffers<T> vec3(&vec2, n1);
-        assert(vec3.eq(&vec1));
-
-        delete _vec2;
+    std::vector<TypeParam*> mem3(end - begin, nullptr);
+    for (int i = 0; i < end - begin; i++) {
+        mem3[i] = mem1->at(i + begin);
     }
+    vec::Buffers<TypeParam> vec3(end - begin, size, &mem3);
 
-    void buffers_utest4()
-    {
-        std::cout << "buffers_utest4\n";
-        int i;
-        int word_size;
+    ASSERT_EQ(vec2, vec3);
+}
 
-        for (i = 0; i <= quadiron::arith::log2<T>(sizeof(T)); i++) {
-            word_size = quadiron::arith::exp2<T>(i);
-            pack_unpack(word_size);
+TYPED_TEST(BuffersTest, Test2) // NOLINT
```

Comment:
@lamphamsy If you have a better name than Test2, I'll be glad to rename it (it was named utest2 before so I kept it like that)

Reply:
It could be renamed as: `Even odd separation`

Reply:
Done